Paintroid/src/main/java/org/catrobat/paintroid/tools/implementation/StampTool.kt

```diff
@@ -0,0 +1,149 @@
+/*
+ * Paintroid: An image manipulation application for Android.
+ * Copyright (C) 2010-2021 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.paintroid.tools.implementation
+
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Color
+import android.os.Bundle
+import org.catrobat.paintroid.R
+import org.catrobat.paintroid.command.CommandManager
+import org.catrobat.paintroid.tools.ContextCallback
+import org.catrobat.paintroid.tools.ToolPaint
+import org.catrobat.paintroid.tools.ToolType
+import org.catrobat.paintroid.tools.Workspace
+import org.catrobat.paintroid.tools.options.StampToolOptionsView
+import org.catrobat.paintroid.tools.options.ToolOptionsVisibilityController
+
+private const val BUNDLE_TOOL_READY_FOR_PASTE = "BUNDLE_TOOL_READY_FOR_PASTE"
+private const val BUNDLE_TOOL_DRAWING_BITMAP = "BUNDLE_TOOL_DRAWING_BITMAP"
+private const val ROTATION_ENABLED = true
+
+class StampTool(
+    stampToolOptionsView: StampToolOptionsView,
+    contextCallback: ContextCallback,
+    toolOptionsViewController: ToolOptionsVisibilityController,
+    toolPaint: ToolPaint,
+    workspace: Workspace,
+    commandManager: CommandManager
+) : BaseToolWithRectangleShape(
+    contextCallback, toolOptionsViewController, toolPaint, workspace, commandManager
+) {
+    private val stampToolOptionsView: StampToolOptionsView
+    private var readyForPaste = false
+    private val isDrawingBitmapReusable: Boolean
+        get() {
+            drawingBitmap?.let {
+                return it.width == boxWidth.toInt() && it.height == boxHeight.toInt()
+            }
+            return false
+        }
+
+    override val toolType: ToolType
+        get() = ToolType.STAMP
+
+    init {
+        rotationEnabled = ROTATION_ENABLED
```

Comment:
I think its better if you just remove ROTATION_ENABLED, and set it true, since its nowhere else used